app/models/user.rb

```diff
@@ -110,4 +113,19 @@ def complete_registration
       registration.organization.update_attribute :active, true if registration.organization.present?
     end
   end
+
+  def technical_user_is_unique_in_org
+    # This validation make the assumption a technical user has only and only one membership at the creation
+    if User.joins(:organizations).where(name: name, 'organizations.id': memberships.first.organization.id).count > 0
```

Comment:
Due to user renaming being possible for both human and technical users, I think the validations should be rewritten to allow creating a human and a technical user with the same name. They will be much simpler that way. It also corresponds to the unique database constraint on the normalized name.

This validation should have `technical: true` added to the `where` clause.

The `validates :name, uniqueness: { unless: : technical }` validation on line 53 should be rewritten as a method, to check only human users.